pkg/server/api.go

```diff
@@ -113,13 +115,14 @@ func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	useragent := r.Header.Get("User-Agent")
-	glog.Infof("Pool %s requested by address:%q User-Agent:%q", cr.machineConfigPool, r.RemoteAddr, useragent)
+	acceptHeader := r.Header.Get("Accept")
+	glog.Infof("Pool %s requested by address:%q User-Agent:%q Accept-Header: %q", cr.machineConfigPool, r.RemoteAddr, useragent, acceptHeader)
 
-	reqConfigVer, err := deriveSpecVersionFromUseragent(useragent)
+	reqConfigVer, err := detectSpecVersionFromAcceptHeader(acceptHeader)
 	if err != nil {
 		w.Header().Set("Content-Length", "0")
 		w.WriteHeader(http.StatusBadRequest)
-		glog.Errorf("invalid useragent: %v", err)
+		glog.Error(err)
```

Comment:
just to confirm this error will be informative enough on its own?

Reply:
yes, it'll only enter here if we get an unsupported ignition version in the accept header. For other headers the default is served.